server/configuration.go

```diff
@@ -109,7 +148,42 @@ func (p *Plugin) OnConfigurationChange() error {
 
 	p.tracker = telemetry.NewTracker(p.telemetryClient, p.API.GetDiagnosticId(), p.API.GetServerVersion(), manifest.Id, manifest.Version, "msteamsmeetings", enableDiagnostics)
 
-	p.setConfiguration(configuration)
+	p.setConfiguration(&loaded)
+
+	// If there was a real change to the configuration, reset all tokens.
+	// Special case the first time OnConfigurationChange is invoked on plugin
+	// load.
+	if (*prev != configuration{}) && *prev != loaded {
```

Comment:
Is this because all configuration settings are related to OAuth? Can we instead explicitly check each individual field to see if it changed, in case we add more plugin settings?